addons/spectrum/functions/fnc_drawSpectrumLine.sqf

```diff
@@ -0,0 +1,56 @@
+#include "script_component.hpp"
+/*/////////////////////////////////////////////////
+Author: Landric
+			   
+File: fnc_drawSpectrumLine.sqf
+Parameters: colour - string corresponding to a colour in CfgMarkerColors
+Return: 
+
+Script to automatically draw a line on the map in the direction they are facing, labelled with the current freq. of the Spectrum Device
+Player must have the Spectrum Device equipped, be in "GUNNER" (i.e. zoomed-in) view, and have a GPS (or UAV terminal) assigned
+
+*///////////////////////////////////////////////
+
+params [["_colour", "ColorBlack", [""]]];
+
+if(!(
+		GVAR(spectrumAutoline) &&
+		(currentWeapon player) isKindOf ["hgun_esd_01_F", configFile >> "CfgWeapons"] &&
+		(
+			(player getSlotItemName 612) isKindOf ["ItemGPS", configFile >> "CfgWeapons"] ||
+			(player getSlotItemName 612) isKindOf ["UavTerminal_base", configFile >> "CfgWeapons"]
+		) &&
+		cameraOn == player && cameraView == "GUNNER"
+	)
+) exitWith { false };
+
+if(!isClass (configFile >> "CfgMarkerColors" >> _colour)) then { _colour = "ColorBlack"; };
+
+private _freq = ((missionnamespace getVariable ["#EM_SelMin", 141.6]) + (missionnamespace getVariable ["#EM_SelMax", 141.9]))/2;
+_freq = (round (_freq*10))/10;
+
+private _marker = createMarkerLocal ["_USER_DEFINED"+(getPlayerUID player)+str(getPos player)+str(getDir player)+str(_freq), player, currentChannel, player];
+_marker setMarkerShapeLocal "polyline";
+_marker setMarkerColorLocal _colour;
+
+
+private _l = 10000; // TODO: base on size of map?
+
+private _pos = [[[position player, GVAR(spectrumAutolineNoise)]]] call BIS_fnc_randomPos;
+
+private _x1 = _pos # 0;
+private _y1 = _pos # 1;
+private _x2 = _x1+(_l*sin(getDir player));
+private _y2 = _y1+(_l*cos(getDir player));
```

Comment:
Alternative option is also the alternative syntax of ``getPos``. 

```sqf
    _dir = direction player; 
    private _nextPos =  player getPos [_length, _dir];
```

As it will return position from player and ``_length`` in the ``_dir`` direction. https://community.bistudio.com/wiki/getPos (syntax 3)

But either works, so just keep it as you have now. It more so you know another time ;-)

Reply:
Well damn, wish I'd spotted that earlier! 😅